pkg/apis/externaldns/types.go

```diff
@@ -578,6 +587,13 @@ func (cfg *Config) ParseFlags(args []string) error {
 	app.Flag("metrics-address", "Specify where to serve the metrics and health check endpoint (default: :7979)").Default(defaultConfig.MetricsAddress).StringVar(&cfg.MetricsAddress)
 	app.Flag("log-level", "Set the level of logging. (default: info, options: panic, debug, info, warning, error, fatal").Default(defaultConfig.LogLevel).EnumVar(&cfg.LogLevel, allLogLevelsAsStrings()...)
 
+	// Plugin provider
+	app.Flag("plugin-provider-url", "[EXPERIMENTAL] The URL of the remote endpoint to call for the plugin provider (default: :8888)").Default(defaultConfig.PluginProviderURL).StringVar(&cfg.PluginProviderURL)
+	app.Flag("run-aws-provider-as-plugin", "[EXPERIMENTAL] When enabled, the AWS provider will be run as a plugin (default: false). To be used together with 'plugin' as provider.").BoolVar(&cfg.RunAWSProviderAsPlugin)
+	app.Flag("plugin-provider-read-timeout", "[EXPERIMENTAL] The read timeout for the plugin provider in duration format (default: 5s)").Default(defaultConfig.PluginProviderReadTimeout.String()).DurationVar(&cfg.PluginProviderReadTimeout)
+	app.Flag("plugin-provider-write-timeout", "[EXPERIMENTAL] The write timeout for the plugin provider in duration format (default: 10s)").Default(defaultConfig.PluginProviderWriteTimeout.String()).DurationVar(&cfg.PluginProviderWriteTimeout)
+	app.Flag("plugin-provider-address", "[EXPERIMENTAL] The address to listen on for the plugin provider (default: :8888)").Default(defaultConfig.PluginProviderAddress).StringVar(&cfg.PluginProviderAddress)
```

Comment:
Should this be restricted to `127.0.0.1:8888` to promote secure defaults?

Reply:
Good point, fixed.